modelcheck/src/main/kotlin/de/itemis/mps/gradle/junit/JUnitXmlSerializer.kt

```diff
@@ -0,0 +1,152 @@
+package de.itemis.mps.gradle.junit
+
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+import java.io.ByteArrayOutputStream
+import java.io.StringWriter
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.Transformer
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+/**
+ * Serializes JUnit test classes to XML format
+ */
+object JUnitXmlSerializer {
+
+    fun toDocument(value: Any): Document =
```

Comment:
This function can be dropped as it adds no real value

Reply:
It lets me avoid duplicated toDocument calls in the writeJUnitXml method, so I can split it into steps: first generate the domain object, then convert it to a document, then write it out.